src/inmanta/deploy/scheduler.py

```diff
@@ -243,7 +245,32 @@ def __init__(
         self.executor_manager = executor_manager
         self._state_update_delegate = ToDbUpdateManager(client, environment)
 
-    def reset(self) -> None:
+        self._sched = Scheduler("Resource scheduler")
+        self.resource_periodic_repairs: dict[ResourceIdStr, ScheduledTask] = {}
+        self._repair_timer: Union[int, str] = agent_config.agent_repair_interval.get()
+
+        self.global_periodic_repair_task: ScheduledTask | None = None
+
+        self.resource_periodic_deploys: dict[ResourceIdStr, ScheduledTask] = {}
+        self._deploy_timer: Union[int, str] = agent_config.agent_deploy_interval.get()
+
+        self.global_periodic_deploy_task: ScheduledTask | None = None
+
+    async def clear_periodic_tasks(self) -> None:
```

Comment:
Why is this async?

I think it shouldn't be, as it creates a race where the timer fires after clearing all other things.

Reply:
Nice catch, this was an artifact from early implementation.